plugins/modules/vios_update_upgrade.py

```diff
@@ -0,0 +1,345 @@
+    elif repo == 'ibmwebsite':
+        mandatoryList += ['image_name']
+        unsupportedList += ['file_list', 'host_name', 'user_id', 'password', 'ssh_key_file', 'mount_loc', 'option', 'directory', 'usb_file_loc']
+    elif repo == 'nfs':
+        mandatoryList += ['mount_loc', 'host_name']
+        unsupportedList += ['user_id', 'password', 'ssh_key_file', 'usb_file_loc']
+    elif repo == 'sftp':
+        mandatoryList += ['user_id', 'host_name']
+        unsupportedList += ['mount_loc', 'option', 'usb_file_loc']
+    elif repo == 'usb':
+        mandatoryList += ['usb_file_loc']
+        unsupportedList += ['file_list', 'host_name', 'user_id', 'password', 'ssh_key_file', 'mount_loc', 'option', 'directory', 'save']
+
+    collate = []
+    for eachUnsupported in unsupportedList:
+        if params[eachUnsupported]:
+            collate.append(eachUnsupported)
+
+    if collate:
+        if len(collate) == 1:
+            raise ParameterError("unsupported parameter: %s" % (collate[0]))
+        else:
+            raise ParameterError("unsupported parameters: %s" % (','.join(collate)))
+        
+    collate = []
+    for eachMandatory in mandatoryList:
+        if not params[eachMandatory]:
+            collate.append(eachMandatory)
+    if collate:
+        if len(collate) == 1:
+            raise ParameterError("mandatory parameter '%s' is missing" % (collate[0]))
+        else:
+            raise ParameterError("mandatory parameters '%s' are missing" % (','.join(collate)))
+
+
+def validate_parameters(params):
+    '''Check that the input parameters satisfy the mutual exclusiveness of HMC'''
+    opr = None
+    unsupportedList = []
+    mandatoryList = []
+
+    if params['state'] is not None:
+        opr = params['state']
+    else:
+        opr = params['action']
+
+    if opr in ['update', 'upgrade']:
+        mandatoryList = ['hmc_host', 'hmc_auth', 'attributes']
+
+    collate = []
+    for eachMandatory in mandatoryList:
+        if not params[eachMandatory]:
+            collate.append(eachMandatory)
+    if collate:
+        if len(collate) == 1:
+            raise ParameterError("mandatory parameter '%s' is missing" % (collate[0]))
+        else:
+            raise ParameterError("mandatory parameters '%s' are missing" % (','.join(collate)))
+    validate_sub_params(params)
+
+def facts(module,params):
+    hmc_host = params['hmc_host']
+    hmc_user = params['hmc_auth']['username']
+    password = params['hmc_auth']['password']
+    validate_parameters(params)
+    attributes = params.get('attributes')
+    hmc_conn = HmcCliConnection(module, hmc_host, hmc_user, password)
+    vios_name = attributes['vios_name'] or attributes['vios_id']
+    m_system = attributes['system_name']
+    sys_list = (
+        hmc_conn.execute("lssyscfg -r sys -F name").splitlines() + hmc_conn.execute("lssyscfg -r sys -F type_model*serial_num").splitlines()
```

Comment:
does this code works if there are no systems connected to HMC or Systems connected to HMC are in non-operation state?

Reply:
If there is no cecs connected to the HMC , then the playbook will return the "The managed system is not available in HMC" error.